AppController/djinn.rb

```diff
@@ -2816,12 +2672,18 @@ def add_routing_for_blob_server(secret)
   #   - NO_HAPROXY_PRESENT: If this node does not run HAProxy (and thus cannot
   #     remove AppServers from HAProxy config files).
   def remove_appserver_from_haproxy(app_id, ip, port, secret)
-    if !valid_secret?(secret)
-      return BAD_SECRET_MSG
-    end
+    return BAD_SECRET_MSG if !valid_secret?(secret)
 
-    if !my_node.is_login?
-      return NO_HAPROXY_PRESENT
+    if !my_node.is_shadow?
```

Comment:
If the shadow roles is becoming the leader, can we change its name to something more meaningful?